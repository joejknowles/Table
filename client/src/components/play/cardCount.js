import React from 'react';
import { connect } from 'react-redux';

import res from '../../resources/pages/play';

import { cardCountSelector } from '../../reducers';

export const CardCount = ({ count }) => (
  <div className="card-count">
    { res.cardCount(count) }
  </div>
);

const mapStateToProps = (state) => ({
  count: cardCountSelector(state)
});

export default connect(mapStateToProps)(CardCount);
