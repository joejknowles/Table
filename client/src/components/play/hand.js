import React from 'react';
import { connect } from 'react-redux';

import PlayCardButton from './playCardButton';
import res from '../../resources/pages/play';

import { myTurnSelector } from '../../selectors';
import { cardCountSelector } from '../../reducers';

export const Hand = ({ myTurn, cardCount }) => (
  <div className="pile">
    { cardCount > 0 ?
      <div className='card'>
        { myTurn && <PlayCardButton /> }
      </div> :
      <p className='no-cards-message'>
        { res.noCardsMessage }
      </p>
    }
  </div>
);

const mapStateToProps = (state) => ({
  cardCount: cardCountSelector(state),
  myTurn: myTurnSelector(state)
});

export default connect(mapStateToProps)(Hand);
