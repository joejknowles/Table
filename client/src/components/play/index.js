import React, { Component } from 'react';
import { connect } from 'react-redux';

import Layout from '../layout'
import Hand from './hand';
//import Snap from './snap';

export class Play extends Component {
  componentWillMount() {
    this.props.ensureConnected();
  }

  render() {
    return (
      <Layout>
        <Hand />
      </Layout>
    );
  }
}

const mapStateToProps = (state, { params: { gameCode } }) => ({
  clientType: state.clientType,
  gameCode
});

const mapDispatchToProps = (dispatch) => ({
  joinAsPlayer: (gameCode) => dispatch({ type: 'PLAYER_JOIN', gameCode })
});

const merge = ({ clientType, gameCode }, { joinAsPlayer }) => {
  const ensureConnected = () => clientType ? null : joinAsPlayer(gameCode);
  return ({
    ensureConnected
  });
};

export default connect(mapStateToProps, mapDispatchToProps, merge)(Play);
