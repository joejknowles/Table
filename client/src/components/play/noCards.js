import React from 'react';
import res from '../../resources/pages/play';

export default () => (
  <div className='pile'>
    <div className="noCards">
        <p className='no-cards-message'>
          { res.noCardsMessage }
        </p>
    </div>
  </div>
);
