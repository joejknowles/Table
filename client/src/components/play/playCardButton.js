import React from 'react';
import { DispatchButton } from '../common/buttons';
import res from '../../resources/pages/play';

export default ({ dispatch }) => (
  <DispatchButton
    res={ res.playCard }
    className='play-card btn'
    action={ { type: 'PLAY_CARD' } }
  />
);
