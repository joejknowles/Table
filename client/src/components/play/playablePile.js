import React from 'react';
import PlayCardButton from './playCardButton';
import SwipeablePile from './swipeablePile';

function is_touch_device() {
  if ( 'callPhantom' in window ) return false; // integration tests fail without this
  return 'ontouchstart' in window ||        // works on most browsers
       navigator.maxTouchPoints;       // works on IE10/11 and Surface
};

export default () => (
    is_touch_device() ?
      <SwipeablePile />
        :
      <PlayCardButton />
);
