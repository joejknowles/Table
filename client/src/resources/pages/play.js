export default {
  playCard: 'Play Card',
  noCardsMessage: 'No more cards'
};
