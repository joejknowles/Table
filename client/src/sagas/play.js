import { call, takeEvery } from 'redux-saga/effects';

import * as toPath from '../routing';
import * as events from '../api/sockets';

export function* playCard(socket, action) {
  yield call(events.playCard, socket, action.card)
}

export function* playerBegin(socket) {
  yield takeEvery('PLAY_CARD', playCard, socket);
  yield call(toPath.play);
}

export function* playerJoin(socket) {
  yield call(events.joinPlayersRoom, socket);
  yield takeEvery('BEGIN_GAME', playerBegin, socket);
  yield call(toPath.waiting);
}
