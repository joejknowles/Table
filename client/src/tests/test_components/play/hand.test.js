import { Hand } from '../../../components/play/hand';

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
const shallowRenderer = ReactTestUtils.createRenderer();

it('Hand renders as before on turn', () => {
  const component = shallowRenderer.render(
    <Hand
      cardCount={ 1 }
      myTurn={ true }
    />
  );
  expect(component).toMatchSnapshot();
});

it('Hand renders as before not on turn', () => {
  const component = shallowRenderer.render(
    <Hand
      cardCount={ 1 }
      myTurn={ false }
    />
  );
  expect(component).toMatchSnapshot();
});

it('Hand renders as before with no cards', () => {
  const component = shallowRenderer.render(
    <Hand
      cardCount={ 0 }
      myTurn={ false }
    />
  );
  expect(component).toMatchSnapshot();
});
