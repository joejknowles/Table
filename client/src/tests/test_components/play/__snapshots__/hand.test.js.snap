exports[`test Hand renders as before not on turn 1`] = `
<div
  className="pile">
  <p
    className="no-cards-message">
    No more cards
  </p>
</div>
`;

exports[`test Hand renders as before on turn 1`] = `
<div
  className="pile">
  <p
    className="no-cards-message">
    No more cards
  </p>
</div>
`;
